### PR TITLE
resource_control: supports dynamically change the controller config

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7043

### What is changed and how does it work?

```commit-message
resource_control: supports dynamically change the controller config 
- supports dynamically changing the controller config
- export the `maxWaitDuration` for the local bucket limiter.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/resource_group/controller/util_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 TiKV Project Authors.
```
Comment (CabinfeverB) [inline client/resource_group/controller/util_test.go:1]: 2023?

**Comment by nolouch** [comment]
PTAL @CabinfeverB @HuSharp

File: client/resource_group/controller/controller.go [hunk 2]
```diff
@@ -213,22 +221,61 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
 			stateUpdateTicker = time.NewTicker(time.Millisecond * 100)
 		})
 
-		_, revision, err := c.provider.LoadResourceGroups(ctx)
+		_, metaRevision, err := c.provider.LoadResourceGroups(ctx)
+		if err != nil {
+			log.Warn("load resource group revision failed", zap.Error(err))
+		}
+		_, cfgRevision, err := c.provider.LoadGlobalConfig(ctx, nil, controllerConfigPath)
```
Comment (HuSharp) [inline client/resource_group/controller/controller.go:228]: I've observed that there are two different approaches to interact etcd. do we need to unify it?
- `LoadGlobalConfig` is impled by pd client
- `Watch` is impled by metaClient

**Review by HuSharp — APPROVED** [review]
LGTM!

File: client/resource_group/controller/controller.go [hunk 3]
```diff
@@ -293,14 +334,35 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
...
+					watchConfigChannel = nil
 					watchRetryTimer.Reset(watchRetryInterval)
 					failpoint.Inject("watchStreamError", func() {
 						watchRetryTimer.Reset(20 * time.Millisecond)
 					})
+					continue
+				}
+				for _, item := range resp {
+					cfgRevision = item.Kv.ModRevision
+					if !strings.HasPrefix(string(item.Kv.Key), controllerConfigPath) {
```
Comment (CabinfeverB) [inline client/resource_group/controller/controller.go:348]: it seems no need to check.

**Review by CabinfeverB — COMMENTED** [review]
REST LGTM

File: server/server.go [hunk 4]
```diff
@@ -1708,7 +1708,10 @@ func (s *Server) campaignLeader() {
 
 	log.Info("triggering the leader callback functions")
 	for _, cb := range s.leaderCallbacks {
-		cb(ctx)
+		if err := cb(ctx); err != nil {
+			log.Error("failed to execute leader callback function", errs.ZapError(err))
+			return
```
[the remark this answers is not in the document]
Comment (nolouch) [reply server/server.go:1713]: I think here some `leaderCallbacks` should make sure to run successfully. like `resourcemanager` load resource meta, or `tso service` init. ptal @rleungx

[the remark this answers is not in the document]
Comment (rleungx) [reply server/server.go:1713]: I think retry is ok to me and we should limit the retry count or time.

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**